packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItemOption/ItemOption.js

```diff
@@ -0,0 +1,119 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+
+import { Search } from '@carbon/react';
+
+import { Checkmark } from '@carbon/react/icons';
+
+import PropTypes from 'prop-types';
+import { ConditionBuilderContext } from '../../ConditionBuilderContext/ConditionBuilderProvider';
+import {
+  blockClass,
+  translateWithId,
+} from '../../ConditionBuilderContext/DataConfigs';
+
+export const ItemOption = ({ conditionState = {}, config = {}, onChange }) => {
+  const { popOverSearchThreshold } = useContext(ConditionBuilderContext);
+  const contentRef = useRef();
+
+  const allOptions = config.options;
+  const [filteredItems, setFilteredItems] = useState(config.options);
+  const selection = conditionState.value;
+  useEffect(() => {
+    //this will focus the first input field in the popover
+
+    if (contentRef.current) {
+      const firstFocusableElement =
+        contentRef.current.querySelector('input, button,li');
+
+      if (firstFocusableElement) {
+        firstFocusableElement.focus();
+      }
+    }
+  }, [allOptions]);
+
+  const onClickHandler = (evt, option) => {
+    onChange(option.id, evt);
+  };
+
+  const onSearchChangeHandler = (evt) => {
+    const { value } = evt.target;
+    let _filteredItems = allOptions.filter((opt) =>
+      opt.label.toLowerCase().includes(value.toLowerCase())
+    );
+    setFilteredItems(_filteredItems);
+  };
+
+  return (
+    allOptions && (
```

Comment:
instead of wrapping all of this in a conditional i think it would be easier and more readable to just define it as

```
if (!alloptions) return;
return...
```

Reply:
modified